app/scripts/clinic-dashboard/directives/patient-flow.directive.js

```diff
@@ -180,17 +198,67 @@
             $scope.averageWaitingTime = results.averageWaitingTime;
             $scope.medianWaitingTime = results.medianWaitingTime;
             $scope.incompleteVisitsCount = results.incompleteVisitsCount;
+
+
             $scope.isBusy = false;
             displayCurrentPatientStatusInfo();
+
+
+          _.each($scope.patientStatuses,function(patient){
```

Comment:
Move this code to it's function. Call the function 'transformVisitsToDummyEncounters' and comment inside the function to explain why we need to do this hack.